### PR TITLE
Run through Active Record Migrations guide

### PR DESCRIPTION
**Pull request by zzak**
Similar to #46700, I'm continuing through the guides this time with [Active Record Migrations](https://guides.rubyonrails.org/active_record_migrations.html).

There are several changes batched up into this PR.

### CODE REVIEW

File: guides/source/active_record_migrations.md [hunk 1]
```diff
@@ -67,22 +67,28 @@ NOTE: There are certain queries that can't run inside a transaction. If your
 adapter supports DDL transactions you can use `disable_ddl_transaction!` to
 disable them for a single migration.
 
+### Making the irreversible possible
```
Comment (p8) [inline guides/source/active_record_migrations.md:70]:
```suggestion
### Making the Irreversible Possible
```

File: guides/source/active_record_migrations.md [hunk 2]
```diff
@@ -101,7 +107,9 @@ class ChangeProductsPrice < ActiveRecord::Migration[7.1]
 end
 ```
 
-Creating a Migration
+INFO: More on [`reversible`](#using-reversible) later.
+
+Generating Migrations
 --------------------
```
Comment (p8) [inline guides/source/active_record_migrations.md:113]:
```suggestion
Generating Migrations
----------------------
```

File: guides/source/active_record_migrations.md [hunk 3]
```diff
@@ -174,42 +184,47 @@ class AddPartNumberToProducts < ActiveRecord::Migration[7.1]
...
   def change
-    remove_column :products, :part_number, :string
+    add_column :products, :part_number, :string
+    add_column :products, :price, :decimal
   end
 end
 ```
 
-You are not limited to one magically generated column. For example:
+### Removing columns
```
Comment (p8) [inline guides/source/active_record_migrations.md:205]:
```suggestion
### Removing Columns
```

File: guides/source/active_record_migrations.md [hunk 4]
```diff
@@ -138,6 +146,8 @@ This generator can do much more than prepend a timestamp to the file name.
 Based on naming conventions and additional (optional) arguments it can
 also start fleshing out the migration.
 
+### Adding new columns
```
Comment (p8) [inline guides/source/active_record_migrations.md:149]:
```suggestion
### Adding New Columns
```

File: guides/source/active_record_migrations.md [hunk 5]
```diff
@@ -174,42 +184,47 @@ class AddPartNumberToProducts < ActiveRecord::Migration[7.1]
...
-    add_column :products, :price, :decimal
+    remove_column :products, :part_number, :string
   end
 end
 ```
 
-If the migration name is of the form "CreateXXX" and is
-followed by a list of column names and types then a migration creating the table
-XXX with the columns listed will be generated. For example:
+### Creating new tables
```
Comment (p8) [inline guides/source/active_record_migrations.md:223]:
```suggestion
### Creating New Tables
```

**Review by p8 — CHANGES_REQUESTED** [review]

**Comment by p8** [comment]
Thanks @zzak!
This was a bigger change to review 😄 
I think most of these are an improvement.

**Comment by zzak** [comment]
Failures are unrelated to this PR

**Pull request merged**